bgpd/bgpd.c

```diff
@@ -202,6 +202,80 @@ int bgp_option_check(int flag)
 	return CHECK_FLAG(bm->options, flag);
 }
 
+/* set the bgp no-rib option during runtime and remove installed routes */
+int bgp_option_norib_set_runtime(void)
+{
+	struct bgp *bgp;
+	struct listnode *node;
+
+	/* check if option is already set in BM */
+	if (bgp_option_check(BGP_OPT_NO_FIB))
+		return 0;
+
+	/* set the no-rib option in BM */
+	bgp_option_set(BGP_OPT_NO_FIB);
+
+	/* informational log message */
+	zlog_info("Disabled BGP route installation to RIB (Zebra)");
+
+	/* iterate through bgp instances */
+	for (ALL_LIST_ELEMENTS_RO(bm->bgp, node, bgp)) {
+		if (!bgp)
```

Comment:
How is this even possible?

Reply:
I don't know. That, and because it is not a performance critical piece of code, is why i opted to waste cycles to check it.

Reply:
I'll expand a bit here on @donaldsharp suprise: when iterating over a list it is unexpected to get a node with `data` pointing to `NULL`, if that is happening we'll much bigger problems than this.

I think it would be best to just drop this conditional and let it crash if we have such problem.